uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -527,6 +533,23 @@ public void rebalanceCurrentCluster() throws Exception {
                 itph.getRoute().getPartition(), expectedNumWorkers - itph.getWorkerSet().size());
           }
         }
+
+        // Expand route when configs are changed
+        if (itph.getWorkerSet().size() < _initMaxNumWorkersPerRoute) {
+          LOGGER.info("Checking route {} with controller {} and topics {} since its number of workers {} "
+                  + "is smaller than initMaxNumWorkersPerRoute {}", itph.getRouteString(), itph.getInstanceName(),
+              itph.getServingTopicPartitionSet(), itph.getWorkerSet().size(), _initMaxNumWorkersPerRoute);
+          int expectedNumWorkers = _initMaxNumWorkersPerRoute;
+          LOGGER.info("current {}, expected {}", itph.getWorkerSet().size(), expectedNumWorkers);
+          if (itph.getWorkerSet().size() < expectedNumWorkers) {
```

Comment:
Why it needs to check again?

Reply:
It's basically the same code as above. Checking again make it the same format as above and  I'll factor out this part later.